src/main/java/Duke.java

```diff
@@ -1,10 +1,406 @@
+     * @param commandArgs description and deadline of the task
+     */
+    private static void executeDeadline(String commandArgs) throws EmptyDescriptionException{
+        if (commandArgs.equals("")) {
+            throw new EmptyDescriptionException();
+        } else {
+            final String[] descriptionAndDeadline = splitDescriptionAndDeadline(commandArgs);
+            Task task = new Deadline(descriptionAndDeadline[0], descriptionAndDeadline[1]);
+            taskList.add(task);
+            getMessageForDeadline(task);
+        }
+    }
+
+
+    private static String[] splitDescriptionAndDeadline(String commandArgs) {
+        return commandArgs.split(" /by ");
+    }
+
+    private static void getMessageForDeadline(Task task) {
+        showAddTask();
+        System.out.println("  " + task.toString());
+        System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+    }
+
+    /**
+     * Add task under task.Event class and feedback display message when task.Event task added
+     * @param commandArgs description of the event
+     */
+    private static void executeEvent(String commandArgs) throws EmptyDescriptionException {
+        if (commandArgs.equals("")) {
+            throw new EmptyDescriptionException();
+        } else {
+            final String[] descriptionAndTime = splitDescriptionAndTime(commandArgs);
+            Task task = new Event(descriptionAndTime[0], descriptionAndTime[1]);
+            taskList.add(task);
+            getMessageForEvent(task);
+        }
+    }
+
+    private static String[] splitDescriptionAndTime(String commandArgs) {
+        return commandArgs.split(" /at ");
+    }
+
+    private static void getMessageForEvent(Task task) {
+        showAddTask();
+        System.out.println("  " + task.toString());
+        System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+    }
+
+    /**
+     * Generate feedback for list of task and task status
+     */
+    private static void executeList() {
+        showTasksList(taskList);
+    }
+
+    private static void showTasksList(ArrayList<Task> taskList) {
+        System.out.println("Here are the tasks in your list:");
+        int taskCount = 1;
+        for (Task task : taskList) {
+            System.out.println(taskCount + "." + task.toString());
+            ++taskCount;
+        }
+    }
+
+    /**
+     * Mark task as done and display message when status of task is changed
+     * @param commandArgs number of the task to be marked as done
+     * @throws EmptyNumberException when no number provided along with done command
+     */
+    private static void executeDone(String commandArgs) throws EmptyNumberException {
+        if (commandArgs.equals("")) {
+            throw new EmptyNumberException();
+        }
+        int taskIndex = Integer.parseInt(commandArgs) - 1; // minus 1 to adhere to array indexing
```

Comment:
Good comment here, without it people would be confused

Another way is to put 1 in a variable named `ARRAY_OFFSET`